### PR TITLE
Added support for time limits

### PR DESCRIPTION
**Pull request by AlannaFey**
Currently I have the max time limit at 10 seconds, if someone submits 100 cases with 10s time limit then that would take like 8 minutes to run so not sure if we want to lower that or find some middle ground with that.  Problems are also going to need a time limit in the database which I'm sure won't be hard to add, but will need to be changed.  Verdict 5 is TLE.

(All the code used is just an empty while loop)
### 1s time limit
image
image

### 10s time limit
image
image

### Time limit gets capped at max (10) seconds
image
image

### Default 1s time limit if none is provided
image
image

### Time limit defaults to 1s, but should be grabbing time limit from database if it exists for problems
image
image

### CODE REVIEW

**Comment by gblake31** [comment]
I think 10 seconds is fine. We have the ability to veto a new problem if it would take way too long to judge.

File: functions/src/judge.ts [hunk 1]
```diff
@@ -39,27 +45,45 @@ async function get_data(problem_id: string): Promise<{
...
 }> {
   const inputs: string[] = []
   const outputs: string[] = []
+  let time_limit: number = DEFAULT_TIME_LIMIT
   // get the data from the database
   await getDoc(doc(db, 'Problems', problem_id))
     .then((problem) => {
       if (problem.exists()) {
         const data = problem.data().sampleData
+        time_limit = data.time_limit
```
Comment (gblake31) [inline functions/src/judge.ts]: It looks like you're using camel case for "sampleData" and snake case for "time_limit." Could you make time_limit camel case to match the format we're using for the database?

image

Comment (gblake31) [inline functions/src/judge.ts]: Right now, you're pulling time_limit from inside sampleData, but I think it should be separate. The sample data is its own thing, separate from the time limit. Not sure if I have the right syntax but could you instead do something like ```problem.data().timeLimit```? DM me this doesn't make sense.

**Comment by gblake31** [comment]
Besides that, looks good! "Adding" timeLimit to the database is super easy, I can do it as soon as this gets merged.

**Comment by AlannaFey** [comment]
> I think 10 seconds is fine. We have the ability to veto a new problem if it would take way too long to judge.

my only concern is arbitrary submissions

**Comment by tylerm390** [comment]
> I think 10 seconds is fine. We have the ability to veto a new problem if it would take way too long to judge.

Are we using pypy or python? If the former 10s should 100% be fine, but Python would be more of a concern. If we aren't using pypy, is there a way to change that?

**Review by gblake31 — DISMISSED** [review]
Looks good, sorry for the delay.

**Review by gblake31 — APPROVED** [review]
It's good to go, make sure to rebase with main!

**Pull request merged**